### ISSUE TITLE
Test on numpy 2.0

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pytensor#688]
### Description

We may need to pin <2.0 until any break changes are fixed (or we could be very lucky and nothing would happen). First step is to test, which we can do with RC versions

**Comment by ricardoV94** [comment]
#689 confirmed we need to pin it. CC @maresb

**Comment by ricardoV94** [comment]
> Any idea on the timeline before the non-rc release drops?

Couldn't find anything conclusive, but sounds like soon: https://github.com/numpy/numpy/issues/24300

**Comment by danieltomasz** [comment]
Here us the issue to track ecosystem compatibility with numpy 2.0 https://github.com/numpy/numpy/issues/26191#issuecomment-2031569341

**Comment by seberg** [comment]
Hmmm, when looking around, I had stumbled on Aesara fork of Theano, but somehow not `pytensor`.  The one potentitially complicated change is that right now I removed `MapIter`, because it is _terribly_ slow (in the public API form) but having it public API means dragging.
Aesara was saying that they wanted to get rid of that code anyway, and I am almost betting that calling back into Python might be better than using `PyArray_MapIter` (e.g. using `ufunc.at()` is usually now fast, but of course not when running on some older NumPy versions)

Please let me know if this is a big issue, or if I can give some pointers in how to just replace the whole thing.

**Comment by jakirkham** [comment]
Is there a NumPy 2 tracking issue for PyMC as well?

Searched for one, but didn't find it (though could very easily have missed it)

**Comment by ricardoV94** [comment]
> Is there a NumPy 2 tracking issue for PyMC as well?
> 
> Searched for one, but didn't find it (though could very easily have missed it)

No, the only blocker for PyMC is PyTensor compatibility with numpy 2.0
